src/context/mixins.ts

```diff
@@ -1,34 +1,58 @@
-/* eslint-disable max-classes-per-file */
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import CryptoJS from "crypto-js";
 
+import { DefaultableMixin } from "../entity/mixins/props";
 import { compareEntitiesInOrderedSetForSorting } from "../entity/set/ordered/utils";
 
-export const ApplicationContextMixin = (superclass) =>
-    class extends superclass {
-        constructor(config) {
-            super(config);
+type Constructor<T = any> = new (...args: any[]) => T;
+
+type Defaultable = ReturnType<typeof DefaultableMixin>;
+
+export function ApplicationContextMixin<T extends Constructor>(superclass: T) {
+    return class ApplicationContextMixin extends superclass {
+        _application: any;
```

Comment:
Is there a way to use an application interface here without importing from `ade`? It's a bit tricky because we monkey patch `Application` and `Material` (`/imports/workflows/context/registry.js`).

Reply:
Added TS interfaces from ESSE